### PR TITLE
PDA-42 process definition list endpoint

### PR DESCRIPTION
**Pull request by ffleandro**
…rver

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: src/main/java/org/entando/plugins/pda/pam/service/process/KieProcessService.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    @Override
+    public List<ProcessDefinition> listDefinitions(Connection connection) {
+        RestTemplate restTemplate = restTemplateBuilder
+                .basicAuthorization(connection.getUsername(), connection.getPassword())
+                .build();
+
+        List<ProcessDefinition> result = new ArrayList<>();
+
+        PagedListRequest pageRequest = new PagedListRequest();
+        pageRequest.setPageSize(2_000_000_000); //Set max page size to get all results
```
Comment (werbth) [inline src/main/java/org/entando/plugins/pda/pam/service/process/KieProcessService.java]: Could you create a constant for this number to improve readability? MAX_PAGE_SIZE maybe?

File: src/test/java/org/entando/plugins/pda/pam/service/process/KieProcessServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        // When
+        List<ProcessDefinition> processDefinitions = kieProcessService.listDefinitions(dummyConnection());
+
+        // Then
+        mockServer.verify();
+        assertThat(processDefinitions).isEqualTo(KieProcessTestHelper.createKieProcessDefinitionList());
+    }
+
+    @Test
+    public void shouldListProcessDefinitionsWithFilterAndSort() throws Exception {
```
Comment (werbth) [inline src/test/java/org/entando/plugins/pda/pam/service/process/KieProcessServiceTest.java]: This test is exactly the same as `shouldListProcessDefinitions`.

File: src/test/java/org/entando/plugins/pda/pam/util/KieProcessTestHelper.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        result.add(KieProcessDefinition.builder()
+                .extraProperty(PROCESS_ID_KEY, PROCESS_DEFINITION_ID_2)
+                .extraProperty(PROCESS_NAME_KEY, PROCESS_NAME_2)
+                .extraProperty(PROCESS_PROP_KEY, PROCESS_PROP_2)
+                .build());
+
+        return result;
+    }
+
+    public static ServiceResponse<KieContainerResourceList> createKieContainersResponse() {
```
Comment (werbth) [inline src/test/java/org/entando/plugins/pda/pam/util/KieProcessTestHelper.java]: This method is not being used.

File: pom.xml [hunk 4]
```diff
@@ -138,6 +143,19 @@
             <version>3.4.1</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
```
Comment (werbth) [inline pom.xml]: Powermock is not being used. Could you remove the dependency, please?

File: pom.xml [hunk 5]
```diff
@@ -108,12 +109,16 @@
...
             <version>1.18.2</version>
         </dependency>
-
         <dependency>
             <groupId>com.github.wnameless.json</groupId>
             <artifactId>json-flattener</artifactId>
             <version>0.7.1</version>
         </dependency>
+        <dependency>
+            <groupId>org.kie.server</groupId>
```
Comment (werbth) [inline pom.xml]: Once you remove unused methods, this dependency can be removed.

**Review by werbth — CHANGES_REQUESTED** [review]

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by werbth** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Review by werbth — APPROVED** [review]

**Pull request merged**